src/Ztm.WebApi/AddressPools/EntityReceivingAddressStorage.cs

```diff
@@ -65,7 +73,12 @@ public async Task<ReceivingAddress> GetAsync(Guid id, CancellationToken cancella
                     .Include(e => e.Reservations)
                     .SingleOrDefaultAsync(r => r.Id == id);
 
-                return recv == null ? null : ToDomain(recv);
+                if (recv == null)
+                {
+                    throw new KeyNotFoundException("The address id is not found.");
+                }
+
+                return ToDomain(recv);
```

Comment:
The previous way should be better for this case due to getting with invalid ID should be common cases.